Documentation/README_Interactable.md

```diff
@@ -2,416 +2,328 @@
+//Access the current DimensionIndex
+int currentDimension = myInteractable.GetDimensionIndex();
 
-Custom Settings added to the new class will also be displayed in the inspector. If there are properties that need to be exposed in the inspector but do not need to be based on states, they should be added to the Custom Settings list.
+//Set the current DimensionIndex to 2
+myInteractable.SetDimensionIndex(2);
 
-### Extending shader themes
+// Promote Dimension to next level
+myInteractable.IncreaseDimension();
+```
 
-If creating a custom interactable theme to modify shader properties, it is advised to extend the [`InteractableShaderTheme`](xref:Microsoft.MixedReality.Toolkit.UI.InteractableShaderTheme) class. Please see the [`InteractableColorTheme`](xref:Microsoft.MixedReality.Toolkit.UI.InteractableColorTheme) class as an example to extend the [`InteractableShaderTheme`](xref:Microsoft.MixedReality.Toolkit.UI.InteractableShaderTheme) class.
+### Create Interactable at runtime
 
-Regardless of whether extending the [`InteractableShaderTheme`](xref:Microsoft.MixedReality.Toolkit.UI.InteractableShaderTheme) class or not, it is highly recommended to set shader properties 1) via [`MaterialPropertyBlock`](https://docs.unity3d.com/ScriptReference/MaterialPropertyBlock.html) and 2) using the shader properties integer keys instead of string keys. Unity assigns integer keys for all shader properties in a project at runtime.See [`Shader.PropertyToID`](https://docs.unity3d.com/ScriptReference/Shader.PropertyToID.html) for more details on. The integer key for an interactable property can be accessed via  `InteractableThemeProperty.GetShaderPropertyId()`.
+*Interactable* can be easily added to any GameObject at runtime. The following example demonstrates how to assign a profile with a [visual theme](visualthemes.md).
 
-Both of these two steps will help with performance. [`MaterialPropertyBlock`](https://docs.unity3d.com/ScriptReference/MaterialPropertyBlock.html) will ensure a new material instance is not created for every object modifying a shader property and integer shader property IDs will eliminate the string-to-int lookup step performed by Unity when using string keys (i.e "_Color") on set/get functions.
+```csharp
+var interactableObject = GameObject.CreatePrimitive(PrimitiveType.Cylinder);
+var interactable = interactableObject.AddComponent<Interactable>();
 
-``` csharp
-public override void SetValue(InteractableThemeProperty property, int index, float percentage)
+interactable.Profiles = new List<InteractableProfileItem>()
 {
-     renderer.GetPropertyBlock(propertyBlock);
-
-     int propId = property.GetShaderPropertyId();
-
-     Color newColor = Color.Lerp(property.StartValue.Color, property.Values[index].Color, percentage);
-     block.SetColor(propId, color);
+    new InteractableProfileItem()
+    {
+        Themes = new List<Theme>() { <insert Theme asset here> },
```

Comment:
Can you actually show how to load / instantiate a theme asset as well? Without that I'm having difficulty creating an interactable at runtime.